sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -668,9 +668,23 @@ def get_throughput(self, **kwargs):
         if response_hook:
             response_hook(self.client_connection.last_response_headers, throughput_properties)
 
+        if 'offerAutopilotSettings' in throughput_properties[0]['content'] and 'autoUpgradePolicy' in \
+                throughput_properties[0]['content']['offerAutopilotSettings']:
+            return ThroughputProperties(properties=throughput_properties[0], auto_scale_max_throughput= \
+                throughput_properties[0]['content']['offerAutopilotSettings']['maxThroughput'],
+                                        auto_scale_increment_percent=throughput_properties[0]['content']
+                                        ['offerAutopilotSettings']['autoUpgradePolicy']['throughputPolicy'][
+                                            'incrementPercent'])
+
+        elif 'offerAutopilotSettings' in throughput_properties[0]['content']:
+            return ThroughputProperties(properties=throughput_properties[0],
+                                        auto_scale_max_throughput=throughput_properties[0]['content']
+                                        ['offerAutopilotSettings']['maxThroughput'])
+
```

Comment:
Okay, yes, three times is two too many.

Reply:
Also to note we are adding the ability to provision autoscale with both containers and databases so this is why you will see this logic in so many different places as we need to provide the logic for both sync and async versions.

Reply:
This logic can and should still be extracted out so that it only exists in one place.

Reply:
Okay sounds good I will get on that then